### PR TITLE
Add txinwitness constructor.

### PR DESCRIPTION
**Pull request by jachiang**
There are a bunch of wrapper classes for the witness, which are just there for serialization, but don't really contribute to the learning process. Suggest to use a txinwitness constructor which consumes an array of witness stack elements.

### CODE REVIEW

**Comment by jachiang** [comment]
Rebased. I have also moved the method into the CTXInWitness class where it makes more sense.

**Comment by jnewbery** [comment]
Thanks for this, @jachiang . Sorry I completely missed it until now (I've been watching the bitcoinops/taproot-workshop repo and not keeping a close eye on this repo). I can either squash these changes into the original PR or you could re-open as a PR against the upstream branch. Whichever you prefer.

File: test_framework/messages.py [hunk 1]
```diff
@@ -381,6 +381,9 @@ def __repr__(self):
     def is_null(self):
         return self.scriptWitness.is_null()
 
+    def from_stack(self, witness_stack):
```
Comment (jnewbery) [inline test_framework/messages.py]: Alternatively, you could just update `__init__()` to optionally take a stack.

Comment (jachiang) [reply test_framework/messages.py]: Agree, that's better. Updated.

**Comment by jnewbery** [comment]
> I can either squash these changes into the original PR or you could re-open as a PR against the upstream branch. Whichever you prefer.

Any preference?

**Comment by jachiang** [comment]
> > I can either squash these changes into the original PR or you could re-open as a PR against the upstream branch. Whichever you prefer.
> 
> Any preference?

Not really, I just rebased it - can you merge it here?

**Comment by jnewbery** [comment]
I've opened a PR against the optech repo here: https://github.com/bitcoinops/taproot-workshop/pull/134

**Pull request closed without merging**